### PR TITLE
opentracing: Update package to use span contexts

### LINKED ISSUE #173: go get github.com/sourcegraph/appdash/opentracing results in error
```
$ go get github.com/sourcegraph/appdash/opentracing
# github.com/sourcegraph/appdash/opentracing
../../sourcegraph/appdash/opentracing/recorder.go:44: sp.Sampled undefined (type basictracer.RawSpan has no field or method Sampled)
../../sourcegraph/appdash/opentracing/recorder.go:49: sp.SpanID undefined (type basictracer.RawSpan has no field or method SpanID)
../../sourcegraph/appdash/opentracing/recorder.go:50: sp.TraceID undefined (type basictracer.RawSpan has no field or method TraceID)
../../sourcegraph/appdash/opentracing/recorder.go:66: sp.Baggage undefined (type basictracer.RawSpan has no field or method Baggage)
```

### PR DESCRIPTION
**Pull request by bg451**
The concept of spancontexts were recently introduced with https://github.com/opentracing/opentracing.github.io/pull/100 and https://github.com/opentracing/basictracer-go/pull/29

Fixes #173 

cc @slimsag @shurcooL

### CODE REVIEW

File: examples/cmd/webapp-opentracing/main.go [hunk 1]
```diff
@@ -110,8 +110,9 @@ func Home(w http.ResponseWriter, r *http.Request) {
 		}
 
 		// We inject the span into the request headers before making the request.
-		carrier := opentracing.HTTPHeaderTextMapCarrier(req.Header)
-		span.Tracer().Inject(span, opentracing.TextMap, carrier)
+
```
Comment (shurcooL) [inline examples/cmd/webapp-opentracing/main.go]: Is this blank line intentional or unintentional?

**Comment by shurcooL** [comment]
The rest looks reasonable to me, but I'm not closely familiar with open tracing API.

The CI is still failing because of some build error in `github.com/influxdata/influxdb/services/httpd`.

**Comment by keegancsmith** [comment]
The influxdb build failure seems to be a problem in upstream influxdb :/ https://github.com/influxdata/influxdb/issues/6924

I'm gonna merge this in just so that we fix build breakages for people not using influx.

**Comment by keegancsmith** [comment]
@bg451 thanks for fixing this!

**Pull request merged**